### PR TITLE
Feature/cross sell logic

### PR DESCRIPTION
**Pull request by cpiehl1**
# Jira Issue: [MX-274] 

## What?
- Move cross-selling logic from giraffe to underwriter

## Why?
- 

## Optional checklist
- [ ] Codescouted
- [x] Unit tests written
- [x] Tested locally

[MX-274]: https://hedvig.atlassian.net/browse/MX-274

### CODE REVIEW

File: src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private fun getSwedishCrossSells(contractTypes: List<TypeOfContract>): List<String> {
+        val isStudent = contractTypes.all { it.isStudentOrYouth() }
+        if (contractTypes.size == 1 && contractTypes.first().isHomeContent()) {
+            return if (isStudent) {
+                listOf("SE_ACCIDENT_STUDENT")
+            } else {
+                listOf("SE_ACCIDENT")
+            }
+        }
```
Comment (mightyost) [inline src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt:32]: I think it is better and clearer if to switch on contract type directly and not have `isHome()` and `isStudentOrYouth()` with its a bit funky logic...

I.e. something like this instead:
```
private fun getSwedishCrossSells(contractTypes: List<TypeOfContract>): List<String> {
        if (contractTypes.size != 1) {
            return listOf()
        }

        when (contractTypes.first()) {
            SE_HOUSE,
            SE_APARTMENT_BRF,
            SE_APARTMENT_RENT -> listOf("SE_ACCIDENT")
            SE_APARTMENT_STUDENT_BRF,
            SE_APARTMENT_STUDENT_RENT -> listOf("SE_ACCIDENT_STUDENT")
            else -> listOf()
        }
}
```

Comment (mightyost) [reply src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt:32]: ...same goes for the other markets...

File: src/main/kotlin/com/hedvig/underwriter/web/ExternalCrossSellController.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@RestController
+@RequestMapping("/crossSells")
+class ExternalCrossSellController(
+    val crossSellService: CrossSellService
+) {
+    @GetMapping
+    @LogCall
+    fun getCrossSells(
+        @RequestHeader("hedvig.token") memberId: String
```
Comment (mightyost) [inline src/main/kotlin/com/hedvig/underwriter/web/ExternalCrossSellController.kt:20]: Nice, this solves the auth problematics!

**Review by mightyost — APPROVED** [review]
Nice job, some cosmetic comments!

File: src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt (hunk 1 above)
Comment (cpiehl1) [reply src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt]: Good point!

Comment (cpiehl1) [reply src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt]: It still became a bit funky with Denmark, where there can be two active contracts, but I think it looks better now 😃

File: src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        return listOf()
+    }
+
+    private fun Contract.isActiveIsh(): Boolean {
+        val activeIshStatus = setOf(
+            ContractStatus.ACTIVE,
+            ContractStatus.ACTIVE_IN_FUTURE,
+            ContractStatus.ACTIVE_IN_FUTURE_AND_TERMINATED_IN_FUTURE,
+            ContractStatus.TERMINATED_TODAY,
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt:88]: Is this Activish?

Comment (meletios) [reply src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt:88]: 😄

Comment (cpiehl1) [reply src/main/kotlin/com/hedvig/underwriter/service/CrossSellServiceImpl.kt:88]: This status was included as `ActiveIsh` in the moving flow, so I felt it made sense to use the same logic.

**Pull request merged**